DIT113-V22/group-01
# 82 add unit tests for quiz logic

Closes #82 

Add all tests for logic of the quiz 
Add corner cases for tests
Update custom quiz class 
Add unit tests to CI

#### Tests are as follows:
- Test chosen category matches categories of all question
- Score increments on question 
- Review setting is done correctly and does not exceed integer limit
- Check validity of needsReview field in all questions
- Check number of questions on choosing question count option
- Check whether all possible outcomes for selecting options (category and number of questions) matches resulting quiz

### :ballot_box_with_check: Definition of Done checklist
- [x] Meaningful title and valuable description in pull request
- [x] Code change tested manually if not covered by automated tests
- [x] Pull request linked to one or more issues/stories
- [x] Acceptance criteria of the linked stories satisfied
- [x] Code change comprehended by those who approved it
- [x] State of target branch maintained or improved after the code change
